solutions/silver/usaco-645.mdx

```diff
@@ -2,159 +2,167 @@
 id: usaco-645
 source: USACO Gold 2016 Open
 title: Splitting the Field
-author: Óscar Garries, Benjamin Qi
+author: Óscar Garries, Benjamin Qi, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_split_gold_open16.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_split_gold_open16.html)
 
-<LanguageSection>
+<Warning title="Weak Test Data">
 
-<CPPSection>
+Some accepted submissions may give different outputs on the following test case:
+
+The right answer should be $6$.
+
+```cpp
+// BeginCodeSnip{Test Case}
+17 1 1 1 2 1 3 2 1 2 2 2 3 3 1 3 2 3 3 3 4 3 5 4 3 4 4 4 5 5 3 5 4 5 5
+    // EndCodeSnip
+```
+</Warning>
+
+## Explanation
+
+Since the enclosures must be rectangular, there *must* be a vertical or horizontal line that separates the two. Othewise,
+we'd have a case where one enclosure has to contain some cow on the x-axis *and* some cow on the y-axis as the other enclosure.
+
+Thus, we'll perform two sweeps, checking all possible vertical and horizontal lines. To keep track of the best answer, we'll keep
+prefix and suffix minimums and maximums to calculate each possible answer in $\mathcal{O}(1)$ time.
+
+It's also possible to use binary search instead of prefix/suffix minimums and maximums.
 
 ## Implementation
 
+**Time Complexity:** $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
 
-const int MX = 5e4;
-int n, x[MX], y[MX], ar[MX];
+using ll = long long;
 
-bool cmp(int a, int b) {
-	if (x[a] == x[b]) return y[a] < y[b];
-	return x[a] < x[b];
-}
+int n;
+ll ans = 0;
+vector<pair<int, int>> cows;
+
+// calculate the maximum area saved by testing splits along cows[i].first
```

Comment:
make this a doc comment
and i think it should return an int and `main()` should update it instead of how it works currently